ming_sim/action_materialize.py

```
@@ -948,6 +948,19 @@ def punish_actions_effective() -> frozenset:
     return punish_actions_allowed() - {"无"}
 
 
+def issue_dispositions_allowed() -> frozenset:
+    """弹劾潮处置枚举唯一真源 = ACTION_CLUSTERS punishment 行。"""
+    cluster = cluster_by_kind("punishment")
+    if cluster is None:
+        raise RuntimeError("punishment cluster not installed")
+    for field in cluster.fields:
```

Ruff (0.16.2)

[error] Import `field` from line 14 shadowed by loop variable

(F402)